src/main/java/com/cognifide/aemrules/Constants.java

```diff
@@ -23,6 +23,8 @@
 
 	public static final String RESOURCE_TYPE = "org.apache.sling.api.resource.Resource";
 
+	public static final String PAGE = "com.day.cq.wcm.api.Page";
```

Comment:
I find this whole class questionable. The naming of these constants is somewhat poor, especially combined with the meaningless class name `Constants`.

I'd go for something along the lines of `SLING_RESOURCE_QUALIFIED_NAME` or `WCM_API_PAGE_QUALIFIED_NAME`

Would it be possible to just replace every occurrence of `Constants.RESOURCE_TYPE` (easy to confuse with the `sling:resourceType` unless you look at the value) with `org.apache.sling.api.resource.Resource.class.getName()`? Or do we want to avoid dependencies on those APIs?

Reply:
I do agree, the class nor naming is not perfect and I'd go with your suggestion. I'd even consider putting `WCM_API_PAGE_QUALIFIED_NAME` into `ContentResourceShouldBeNullCheckedCheck` as it is has one & only usage right now.

Regarding dependencies, this unfortunately would mean we have to package these dependencies into plugin, which seems to be a bad idea.

Reply:
done